main.py

```diff
@@ -179,24 +179,18 @@ def chooser_callback(self,uri_list):
         '''
         try:
             ss = SharedStorage()
-            for uri in uri_list: # only one list
-                
-                # copy to private cache
-                path = ss.copy_from_shared(uri)
-
-                #read content from cache
-                try:
-                    with open(path,'r',encoding = 'utf-8') as fread:
-                        reading = fread.read()
-
-                    #Clock is needed otherwise throw thread error
-                    Clock.schedule_once(lambda dt: self.populate_label(reading))
+            # copy to private cache
+            path = ss.copy_from_shared(uri[0])
```

Comment:
@javixav I believe the uri index has been accessed when `self.chooser_callbac` has been called from the `open_file` method. So the value of uri should be a string not a list. So I think you should remove the `[0]` and just do this `path = ss.copy_from_shared(uri[0])